### DISCUSSION TITLE
How can I get update of someone when they add the bot to group

### DISCUSSION THREAD

**Discussion by foxdevuz** [nutgram/nutgram#805] [Q&A] [answered]
how can I get the update when someone added bot to the group
so far I tried this but it didn't worked on me:

```
$bot->onChatMember(function (Nutgram $bot, ChatMemberUpdated $chatMemberUpdated) {
    if ($chatMemberUpdated->new_chat_member->user->is_bot &&
        $chatMemberUpdated->new_chat_member->user->id === $bot->getMe()->id) {
        $chatId = $chatMemberUpdated->chat->id;
        
        $bot->sendMessage(
            text: "Hello, group! I'm your new bot.",
            chat_id: $chatId
        );
    }
});
```

I got this from [docs](https://nutgram.dev/docs/usage/handlers): 

> onChatMember($callable)
Handles any chat member in other chats when updated.

but once I tried it didn't work.

can anyone please help? 

Thanks!

**Comment by Lukasss93** [comment, marked as answer]
The `onChatMember` method is a listener that listens for new chat members. (you need to be an admin to listen to this event)
Source: https://core.telegram.org/bots/api#update

You need to use the `onMyChatMember` listener:

```php
$bot = new Nutgram('your-token');

$bot->onMyChatMember(function (Nutgram $bot) {
    $newChatMember = $bot->chatMember()->new_chat_member;
    
    if ($newChatMember->user->is_bot && $newChatMember->user->id === $bot->getMe()->id) {
        $bot->sendMessage(
            text: "Hello, group! I'm your new bot.",
        );
    }
});

$bot->run();
```

**Reply by foxdevuz** [reply]
Thank you so much
